src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -289,14 +442,44 @@ public static partial class JsonSerializer
             }
         }
 
+        internal static TValue? ReadAll<TValue>(
```

Comment:
nit: for consistency with other read/write helpers that use `JsonTypeInfo` (or remove suffix from the existing ones)
```suggestion
        internal static TValue? ReadAllUsingMetadata<TValue>(
```

Reply:
Hmm I think removing suffix is better - otherwise we'd have "ReadAllUsingMetadataAsync" which is a little verbose, plus the "Metadata" nomenclature is lost right after that (everything becomes metadata-based at runtime).